exodus_core/analysis/static_analysis.py

```diff
@@ -323,14 +325,52 @@ def _get_icon_from_details(self, path):
 
         raise Exception('Unable to download the icon from details')
 
-    def _get_icon_from_gplay(self, handle, path):
+    def _get_icon_from_store(self, path, source):
+        if source == "fdroid":
+            saved_path = self._get_icon_from_fdroid(path)
+        else:
+            saved_path = self._get_icon_from_gplay(path)
+
+        return saved_path
+
+    def _get_icon_from_fdroid(self, path):
+        """
+        Download the application icon from F-Droid website
+        :param path: file to be saved
+        :return: path of the saved icon
+        :raises Exception: if unable to download icon
+        """
+        handle = self.get_package()
+        index_file_path = '{}/.exodus/index.xml'.format(Path.home())
```

Comment:
You may should use Minio storage instead of local FS. Exodus worker can run on different and multiple hosts at the same time, so, to prevent downloading the F-Droid repository multiple times on the different hosts you should prefer Minio.